src/agent/src/mount.rs

```diff
@@ -767,10 +771,10 @@ fn ensure_destination_file_exists(path: &Path) -> Result<()> {
         return Err(anyhow!("{:?} exists but is not a regular file", path));
     }
 
-    // The only way parent() can return None is if the path is /,
-    // which always exists, so the test above will already have caught
-    // it, thus the unwrap() is safe
-    let dir = path.parent().unwrap();
+    let dir = path
+        .parent()
+        .ok_or(format!("failed to find parent path for {:?}", path))
+        .map_err(|e| anyhow!(e))?;
```

Comment:
As noted in the comment, we've already tested for the failure case, so `unwrap()` is reasonable here.